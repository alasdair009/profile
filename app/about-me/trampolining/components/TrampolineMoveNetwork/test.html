<html lang="en">
  <head>
    <style>
      .videos {
        display: flex;
        height: 100vh;
        width: 100vw;
      }
      .videos div {
        align-items: center;
        display: flex;
        flex-direction: column;
        justify-content: center;
        width: 50%;
      }
      iframe {
        aspect-ratio: 16/9;
        width: 90%;
      }
    </style>
  </head>
  <body>
    <div class="videos">
      <div>
        <iframe
          id="standard"
          src=""
          data-src="https://www.youtube.com/embed/NpEaa2P7qZI?si=uJjK4C2TLWivF5BX"
          title="Standard Embed"
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
          referrerpolicy="strict-origin-when-cross-origin"
          allowfullscreen
        ></iframe>
        <button id="goStandard">Standard</button>
      </div>
      <div>
        <iframe
          id="privacy"
          src=""
          data-src="https://www.youtube-nocookie.com/embed/NpEaa2P7qZI?si=uJjK4C2TLWivF5BX"
          title="Enhanced privacy"
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
          referrerpolicy="strict-origin-when-cross-origin"
          allowfullscreen
        ></iframe>
        <button id="goPrivacy">Privacy</button>
      </div>
    </div>
    <script>
      const goStandard = document.getElementById("goStandard");
      const standardIFrame = document.getElementById("standard");
      const goPrivacy = document.getElementById("goPrivacy");
      const privacyIFrame = document.getElementById("privacy");
      goStandard.addEventListener(
        "click",
        function () {
          standardIFrame.setAttribute("src", standardIFrame.getAttribute("data-src"));
        },
        false
      );
      goPrivacy.addEventListener(
        "click",
        function () {
          privacyIFrame.setAttribute("src", privacyIFrame.getAttribute("data-src"));
        },
        false
      );
    </script>
  </body>
</html>
