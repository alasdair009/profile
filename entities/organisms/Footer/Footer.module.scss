@use "../../../app/styles/animation";
@use "../../../app/styles/colors";
@use "../../../app/styles/dimensions";

@keyframes waving {
  0% {
    transform: translate3d(-90px, 0, 0);
  }
  100% {
    transform: translate3d(85px, 0, 0);
  }
}

.root {
  margin: 0 auto;
  position: relative;
  width: 100%;
}

.waves {
  position: relative;
  width: 100%;
  margin-bottom: -7px; /*Fix for safari gap*/
  height: dimensions.getSize(s128);
}

.parallax {
  > use {
    animation: waving animation.animationDuration(25)
      cubic-bezier(0.55, 0.5, 0.45, 0.5) infinite;
  }
  > use:nth-child(1) {
    animation-delay: -2s;
    animation-duration: 7s;
  }
  > use:nth-child(2) {
    animation-delay: -3s;
    animation-duration: 10s;
  }
  > use:nth-child(3) {
    animation-delay: -4s;
    animation-duration: 13s;
  }
  > use:nth-child(4) {
    animation-delay: -5s;
    animation-duration: 20s;
  }
}

.wave1 {
  fill: darken(colors.$colorGreenGrass, 30%);
}

.wave2 {
  fill: darken(colors.$colorGreenGrass, 25%);
}

.wave3 {
  fill: darken(colors.$colorGreenGrass, 15%);
}

.wave4 {
  fill: colors.$colorGreenGrass;
}
