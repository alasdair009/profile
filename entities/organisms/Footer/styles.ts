"use client";

import styled, { keyframes } from "styled-components";
import { colors, sizes } from "@/entities";
import { darken } from "polished";

export const footerHeight = sizes.s128.rem;

const waving = keyframes`
    0% {
        transform: translate3d(-90px,0,0);
    }
    100% {
        transform: translate3d(85px,0,0);
    }
`;

export const Root = styled.footer`
  margin: 0 auto;
  position: relative;
  width: 100%;
`;

export const Waves = styled.svg`
  position: relative;
  width: 100%;
  margin-bottom: -7px; /*Fix for safari gap*/
  height: ${sizes.s128.rem};
`;

export const Parallax = styled.g`
  > use {
    animation: ${waving} 25s cubic-bezier(0.55, 0.5, 0.45, 0.5) infinite;
  }
  > use:nth-child(1) {
    animation-delay: -2s;
    animation-duration: 7s;
  }
  > use:nth-child(2) {
    animation-delay: -3s;
    animation-duration: 10s;
  }
  > use:nth-child(3) {
    animation-delay: -4s;
    animation-duration: 13s;
  }
  > use:nth-child(4) {
    animation-delay: -5s;
    animation-duration: 20s;
  }
`;

export const Wave1 = styled.use`
  fill: ${darken(0.5, colors.greenGrass)};
`;

export const Wave2 = styled.use`
  fill: ${darken(0.25, colors.greenGrass)};
`;

export const Wave3 = styled.use`
  fill: ${darken(0.15, colors.greenGrass)};
`;

export const Wave4 = styled.use`
  fill: ${colors.greenGrass};
`;
