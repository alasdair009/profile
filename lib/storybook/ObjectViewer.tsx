import { Source } from "@storybook/blocks";

import type { JSX } from "react";

export type ObjectViewerProps = {
  /**
   * The object to display
   */
  object: object;
  /**
   * The title of the code segment
   */
  title: string;
};

export function ObjectViewer({
  object,
  title,
}: ObjectViewerProps): JSX.Element {
  // Format the object as JSON with indentation
  const objectAsString = JSON.stringify(object, null, 4);

  return (
    <div title={title} data-testid={ObjectViewer.name}>
      <Source code={objectAsString} dark={true} />
    </div>
  );
}
