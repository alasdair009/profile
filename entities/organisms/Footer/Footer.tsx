import { HTMLAttributes } from "react";
import { Parallax, Root, Wave1, Wave2, Wave3, Wave4, Waves } from "./styles";

type FooterProps = {} & HTMLAttributes<HTMLDivElement>;
export function Footer({ ...rest }: FooterProps) {
  return (
    <Root data-testid={Footer.name} {...rest}>
      <Waves
        xmlns="http://www.w3.org/2000/svg"
        xmlnsXlink="http://www.w3.org/1999/xlink"
        viewBox="0 24 150 28"
        preserveAspectRatio="none"
        shape-rendering="auto"
      >
        <defs>
          <path
            id="gentle-wave"
            d="M-160 44c30 0 58-18 88-18s 58 18 88 18 58-18 88-18 58 18 88 18 v44h-352z"
          />
        </defs>
        <Parallax>
          <Wave1 xlinkHref="#gentle-wave" x="48" y="0" />
          <Wave2 xlinkHref="#gentle-wave" x="48" y="3" />
          <Wave3 xlinkHref="#gentle-wave" x="48" y="5" />
          <Wave4 xlinkHref="#gentle-wave" x="48" y="7" />
        </Parallax>
      </Waves>
    </Root>
  );
}
