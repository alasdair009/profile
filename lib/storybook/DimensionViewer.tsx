import { Source } from "@storybook/blocks";
import type { HTMLAttributes } from "react";

import { ObjectViewer } from "./ObjectViewer";
import { colors } from "@/entities";

export type DimensionViewerProps = {
  /**
   * The units to display
   */
  units: Record<string, unknown>;
  /**
   * A simplified object that can be used for the DimensionBar interface
   */
  barUnits: Record<string, string | number>;
  /**
   * The title of the code segment
   */
  title: string;
  /**
   * The suffix to display after the unit
   */
  suffix?: string;
  /**
   * Source example
   */
  sourceExample: string;
  /**
   * The name of the object
   */
  objectName: string;
};

export function DimensionViewer({
  units,
  title,
  sourceExample,
  objectName,
  barUnits,
  suffix = "",
}: DimensionViewerProps): JSX.Element {
  return (
    <div data-testid={DimensionViewer.name}>
      <ObjectViewer object={units} title={objectName} />
      <p>
        To use these values import the <strong>{objectName}</strong> object and
        reference the required key.
      </p>
      <Source code={sourceExample} dark={true} language="css" />
      <DimensionBarWrapper title={title}>
        {Object.keys(barUnits).map((key, index) => {
          return (
            <DimensionBar
              key={`${title}${index}`}
              data-testid={DimensionViewer.name}
              width={`${barUnits[key]}${suffix}`}
            >
              <DimensionBarLabel>
                {barUnits[key]}
                {suffix}
                <br />({key})
              </DimensionBarLabel>
            </DimensionBar>
          );
        })}
      </DimensionBarWrapper>
    </div>
  );
}

const DimensionBarWrapper = ({ children }: HTMLAttributes<HTMLDivElement>) => (
  <div
    style={{
      display: "flex",
      flexDirection: "column",
      overflow: "scroll",
      width: "100%",
    }}
  >
    {children}
  </div>
);

const DimensionBar = ({
  children,
  width,
}: { width: string } & HTMLAttributes<HTMLDivElement>) => (
  <div
    style={{
      background: colors.greenGrass,
      color: "white",
      flex: 1,
      marginBottom: "8px",
      width,
    }}
  >
    {children}
  </div>
);

const DimensionBarLabel = ({ children }: HTMLAttributes<HTMLSpanElement>) => (
  <span
    style={{
      display: "block",
      marginLeft: "8px",
    }}
  >
    {children}
  </span>
);
