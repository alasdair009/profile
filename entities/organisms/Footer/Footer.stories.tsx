import { Footer } from "./Footer";
import { Meta, StoryObj } from "@storybook/nextjs";
import { expect, within } from "storybook/test";

const meta: Meta<typeof Footer> = {
  component: Footer,
  parameters: {
    controls: { hideNoControlsWarning: true },
  },
};
export default meta;

export const Default: StoryObj<typeof Footer> = {
  play: async ({ args, canvasElement }) => {
    const canvas = within(canvasElement);
    const footerElement = canvas.getByTestId(Footer.name);

    await expect(footerElement).toBeInTheDocument();
  },
};
