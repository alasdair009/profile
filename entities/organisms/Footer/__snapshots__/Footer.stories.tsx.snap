// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`organisms/Footer Default play-test 1`] = `
<footer data-testid="Footer"
        class="sc-dKsqdn bqWmUS"
>
  <svg xmlns="http://www.w3.org/2000/svg"
       xmlns:xlink="http://www.w3.org/1999/xlink"
       viewbox="0 24 150 28"
       preserveaspectratio="none"
       shape-rendering="auto"
       class="sc-eowDPD dZMiOJ"
  >
    <defs>
      <path id="gentle-wave"
            d="M-160 44c30 0 58-18 88-18s 58 18 88 18 58-18 88-18 58 18 88 18 v44h-352z"
      >
      </path>
    </defs>
    <g class="sc-iAlELC kDzrcj">
      <use xlink:href="#gentle-wave"
           x="48"
           y="0"
           class="sc-oeqTF eaZGlS"
      >
      </use>
      <use xlink:href="#gentle-wave"
           x="48"
           y="3"
           class="sc-ezTrPE braYQM"
      >
      </use>
      <use xlink:href="#gentle-wave"
           x="48"
           y="5"
           class="sc-drnuxz ijdAcY"
      >
      </use>
      <use xlink:href="#gentle-wave"
           x="48"
           y="7"
           class="sc-hDcvty jFZNhr"
      >
      </use>
    </g>
  </svg>
</footer>
`;
